app/assets/javascripts/checkbox.js

```diff
@@ -0,0 +1,23 @@
+function selectChildren(select_all_element,nesting_level){
+    let children_checkboxes = $j(':checkbox', $j(select_all_element).parents(":eq("+nesting_level+")"))
```

Comment:
The use of parents(n) makes it quite "powerful" and simple to use... 
I agree that it can be "broken", but for it to be interfered with, you'd actually need to create a nesting level between the place where this function is called and the objective... 
In the split buttons, for example, you'd need to create a div around them *only* (i.e. not the item), which does not seem so easy to do without noticing you've broken something...
Also, fixing it when broken is incredibly simple, you just need to increase/decrease the nesting_level.

Passing a selector could also work, but in its current uses it means transferring that selector across a few partials, as locals, and sometimes these "variables" need to be used in js, which complicates matters a bit... one could also argue that a selector could be broken too, for example if the name of the referenced item is modified...
So I feel inclined to leave this as it is, but if your experience tells you think I really should change this to a reference pass, then I will do so...

Reply:
You can still use `parents` (or `closest`), but pass a selector with something meaningful, such as a class name or data attribute (we use this in other places in SEEK https://github.com/seek4science/seek/blob/b70353488062a903c47bdbe7935b078796fc13d4/app/assets/javascripts/application_shared.js.erb#L194), so it's explicit and easier to see where the selection is applied. Feel free to add/modify existing page elements to make them easier to select.

The way it is currently, people wanting to use this method will have to count parent elements, which is not always straightforward if the checkboxes are rendered dynamically via JavaScript or something.

Reply:
Resolved in [28b5992](https://github.com/seek4science/seek/pull/1399/commits/28b59925156c7eb8da688d76dd76e65eb1642aa1).